### DISCUSSION TITLE
GDScript Variables References

### DISCUSSION THREAD

**Discussion by dxur** [godotengine/godot-proposals#8133] [Scripting]
### Describe the project you are working on

Implementing something similar to references in c++ then you will be able to set variable value from its reference.

# Syntax

`var int_ref:Ref[float]`
`int_ref = &int_var `
`# or `
`int_ref = ref_of(int_var)`

### Describe the problem or limitation you are having in your project

N/A

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

N/A

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

N/A

### If this enhancement will not be used often, can it be worked around with a few lines of script?

N/A

### Is there a reason why this should be core and not an add-on in the asset library?

THIS NEED TO BE IMPLEMENTED ON THE GDSCRIPT PARSER.

**Comment by Calinou** [comment]
Can you give any concrete use cases for this feature? Most scripting languages don't feature pointers/references like this.

**Reply by dxur** [reply]
this will help editing values from the function "as passing a reference" , then i have started implementing this as my college uses GDscript and godot to test AI functions "as godot 4 support compute shaders and gdextention", then i think it will be a good enhancement making the code more readable ...
Then I'm asking for if is it worth it spend some time implementing it, then i don't need a lot of help i know what I'm doing...

**Reply by Calinou** [reply]
What I mean is, why would AI need references instead of variables that are passed by value? As far as I know, there isn't anything in programming that *requires* references to be available in the language.

**Reply by dxur** [reply]
Not just for AI, you know that GdScript support references for objects this will be for variables and that set. It doesn't need more explanation. Anyway, I will implement this then I'll be happy to add this to the official release

**Reply by dxur** [reply]
Then references are just the safe version of pointers.

**Reply by KoBeWi** [reply]
C# supports variable references using `ref`. There was a proposal to make use of it in Tweens: https://github.com/godotengine/godot-proposals/issues/3220
I think this feature would be a nice complement for Callables. It would allow to fix some methods that still use strings.
